File: client/src/app/shared/models/product.ts
import { Currency } from './currency';

export interface IProduct {
  id: number;
  title: string;
  thumbnailURL: string;
  description: string;
  price?: number | null;
  priceString?: string | null;
  currency: Currency | number;
  isBoosted: boolean;
  urlToProduct: string;
  isGood: boolean;
}

File: Core/Entities/Product.cs
using static Core.Enums.Enum;

namespace Core.Entities
{
    public class Product : BaseEntity
    {
        public string Title { get; set; }
        public string ThumbnailURL { get; set; }
        public string Description { get; set; }
        public int? Price { get; set; }
        public string PriceString { get; set; }
        public Currencies Currency { get; set; }
        public bool IsBoosted { get; set; }
        public string UrlToProduct { get; set; }
        public bool IsGood { get; set; }
    }
}